scripts/calculate_checksum.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+"""
+Script to generate firmware integrity information for the STM32 module
+startup application.
+
+The script takes an Intel Hex file as an input, and parses it to generate
+integrity information that can then be loaded into a binary using objcopy.
+
+"""
+import argparse
+from dataclasses import dataclass
+import re
+import zlib # for crc calc
+from pathlib import Path
+
+# REGEX to split a line in hex file.
+
+
+INITIAL_CRC = 0xFFFFFFFF
+
+class HexRecord:
+    """Represents a single record in a hex file."""
+    PATTERN = re.compile(r"^:([A-F0-9]{2})([A-F0-9]{4})(\d{2})([A-F0-9]*)([A-F0-9]{2})$")
+
+    _SEGMENT_ADDR_TYPE = 2
+    _LINEAR_ADDR_TYPE = 4
+
+    def __init__(self, line: str):
```

Comment:
I think this works, but I will say that you could run `stm32-tools/gcc-arm-embedded/${PLATFORM}/bin/arm-none-eabi-objcopy -Iihex -Obinary --gap-fill=0xff` and then checksum that rather than writing your own hex decoder

Reply:
Yeah that makes more sense. [Pain](https://en.wikipedia.org/wiki/Sunk_cost)

Reply:
Switched over to checking .bin files, and verified that the CRC and length stay the same between the two methods